src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -907,7 +907,9 @@ private Expr createFieldsExtractorClassInstance(
       for (int i = 0; i < descendantFields.length; i++) {
         String currFieldName = descendantFields[i];
         String bindingFieldMethodName =
-            String.format("get%s", JavaStyle.toUpperCamelCase(currFieldName));
+            (i < descendantFields.length - 1 || !httpBindingFieldName.isRepeated())
```

Comment:
Is it easier to read if we change it to `httpBindingFieldName.isRepeated() && i == descendantFields.length - 1`? Also is it a valid case that a repeated field appear in positions other than the last one? Or you are just trying to be defensive here?
If it is indeed a valid case, can you please add a test case for it as well?

Reply:
I just keep `i < descendantFields.length - 1` construct in the if statement to be consistent with the rest of similar statements in the same method (there are several of them in this method). 

No, repeated field can't have "nested" fields because repeated field is a List. But most importantly `httpBindingFieldName.isRepeated()` always represents the characteristic of the last field in the `.` chain only. I.e. if we have `field_one.field_two.field_three` there will be only one binding object with `name = "field_one.field_two.field_three"` and `isRepeated = false` if `field_three` is a regular field and `isRepeated = true` if it is repeated. `field_one` and `field_two` are guaranteed to be non-primitive, non-repeated message fields (when the type of `field_one` has a field named `field_two`), otherwise it would be a broken API definition.

Reply:
If it's not a valid case, can we simplify it to `if (!httpBindingFieldName.isRepeated())`?